### PR TITLE
colour adjacent face bug [closes issue 1033]

### PR DESCRIPTION
**Pull request by rupertford**

### CODE REVIEW

**Comment by rupertford** [comment]
At the moment the symbol table is not being used for the variables `cell`, `colour` or `cmap`. I therefore continued to use strings in this PR as moving to using a symbol table would have involved changing additional classes and adding new tests which would have taken the changes beyond the purpose of this PR.

**Comment by rupertford** [comment]
OK, ready for first review. Probably from @arporter or @TeranIvy

**Comment by TeranIvy** [comment]
I am happy to take this and I will ask Christine to try it.

File: src/psyclone/tests/domain/lfric/transformations/dynamo0p3_transformations_test.py [hunk 1]
```diff
@@ -293,6 +293,39 @@ def test_colour_trans_stencil(dist_mem, tmpdir):
...
+        index = 0
+
+    # Colour the loop
+    ctrans.apply(schedule.children[index])
+
+    # Store the results of applying this code transformation as
+    # a string
+    gen = str(psy.gen)
+
+    # Check that we index the stencil dofmap appropriately
```
Comment (TeranIvy) [inline src/psyclone/tests/domain/lfric/transformations/dynamo0p3_transformations_test.py:320]: s/stencil dofmap/adjacent face dofmap/g

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Nice simple PR. Looks good and produces correct code in LFRic.
- Changes are covered with tests and are `pycodestyle` and `pylint` clean.
- Tests with compilation pass.
- Examples run and documentation builds.

 There is only one inline remark and I would also kindly ask the developer to bring the branch up-to-date with the master (including the recent `fparser` update). Apart from that we are ready to go.

File: src/psyclone/tests/domain/lfric/transformations/dynamo0p3_transformations_test.py (hunk 1 above)
Comment (rupertford) [reply src/psyclone/tests/domain/lfric/transformations/dynamo0p3_transformations_test.py:320]: Oops, thanks.

**Comment by rupertford** [comment]
Brought the branch up-to-date and addressed the reviewers comments. Ready for next review subject to GHA being happy.

**Review by TeranIvy — APPROVED** [review]
Thanks for making changes! All well now. Proceeding to merge.

**Pull request merged**